src/Layout.js

```diff
@@ -1506,13 +1541,13 @@ var computeLayout = (function() {
 
       // First, try to use the layout cache.
       if (layout.cachedLayout &&
-          canUseCachedMeasurement(availableWidth, availableHeight, marginAxisRow, marginAxisColumn,
+          canUseCachedMeasurement(node.isTextNode, availableWidth, availableHeight, marginAxisRow, marginAxisColumn,
```

Comment:
There are some optimizations which are specific to how text is measured which might not apply to all height-for-width measured nodes. This makes it easier to think about and more clear where this optimization will work correctly.